index.bs

```diff
@@ -188,8 +188,11 @@ The requestStorageAccessForOrigin API defines a [=powerful feature=] identified
   <dd>
     <div algorithm='top-level-storage-access-key-generation'>
     To generate a new [=permission key=] for the "<a permission><code>top-level-storage-access</code></a>" feature, given an [=environment settings object=] |settings|, run the following steps:
+    1. Let |current origin| be |settings|' [=environment settings object/origin=].
+    1. If |current origin| is not [=same site=] with |settings|' [=top-level origin=], return null.
     1. Return the result of [=obtain a site|obtaining a site=] from |settings|' [=top-level origin=].
 
+    Note: the check for whether |settings|' [=environment settings object/origin=] is [=same site=] with |settings|' [=top-level origin=] is intended to disallow permission queries from cross-site frames.
```

Comment:
Can you add some more text here about how we maintain the invariant that this will only ever be used in permissions.query, and never when storing new permissions, because requesting permission is only possible from a top-level context?